treeherder/seta/preseed.py

```diff
@@ -13,30 +13,32 @@
 def load_preseed():
     """ Update JobPriority information from preseed.json
 
-    We currently call this method on a schedule. See treeherder/seta/tasks.py for details.
-
     The preseed data has these fields: buildtype, testtype, platform, priority, expiration_date
     The expiration_date field defaults to 2 weeks when inserted in the table
     The expiration_date field has the format "YYYY-MM-DD", however, it can have "*" to indicate to never expire
     The default priority is 1, however, if we want to force coalescing we can do that
     The fields buildtype, testtype and platform can have * which makes ut match  all flavors of
     the * field. For example: (linux64, pgo, *) matches all Linux 64 pgo tests
     """
-    preseed = preseed_data()
-    for job in preseed:
-        queryset = JobPriority.objects.all()
-
-        for field in ('testtype', 'buildtype', 'platform'):
-            if job[field] != '*':
-                queryset = queryset.filter(**{field: job[field]})
-
-        # Deal with the case where we have a new entry in preseed
-        if len(queryset) == 0:
-            create_new_entry(job)
-        else:
-            # We can have wildcards, so loop on all returned values in data
-            for jp in queryset:
-                process_job_priority(jp, job)
+
+    if JobPriority.objects.count() == 0:
```

Comment:
Yeah I almost suggested that but the docs implied that `.exists()` might evaluate the queryset and so not be as optimal as `.count()` which it described as doing a `SELECT COUNT(*)`.

However your comment made me curious and in fact it appears the optimiser does the right thing for `.exists()` in this case...

```
$ ./manage.py shell_plus --print-sql
...
>>> JobPriority.objects.count()
SELECT COUNT(*) AS `__count`
FROM `seta_jobpriority`

Execution time: 0.000573s [Database: default]

>>> JobPriority.objects.exists()
SELECT (1) AS `a`
FROM `seta_jobpriority`
LIMIT 1

Execution time: 0.000752s [Database: default]
```